### PR TITLE
Anchor slide & scroll button

### PR DESCRIPTION
**Pull request by bso-odoo**

### CODE REVIEW

File: addons/website/static/src/interactions/anchor_slide.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    static dynamicContent = {
+        "_root": {
+            "t-on-click": "animateClick",
+        },
+    };
+    
+    /**
+     * @param {DOMElement} el the element to scroll to.
+     * @param {string} [scrollValue='true'] scroll value
+     * @returns {Promise}
```
Comment (ged-odoo) [inline addons/website/static/src/interactions/anchor_slide.js:16]: does not return promise

File: addons/website/static/tests/interactions/anchor_slide.test.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        <a id="somewhere" />
+      </div>
+    `);
+    expect(core.interactions.length).toBe(0);
+});
+
+test("anchor slide scrolls to targetted location", async () => {
+    let didScroll = false;
+    const oldToScroll = AnchorSlide.prototype.toScroll;
+    patchWithCleanup(AnchorSlide.prototype, {
```
Comment (ged-odoo) [inline addons/website/static/tests/interactions/anchor_slide.test.js]: just to be sure, `patchWithCleanup(window, { scrollTo() { ...} });` does not work?

File: addons/website/static/src/snippets/s_table_of_content/000.js [hunk 3]
```diff
@@ -222,25 +224,19 @@ const TableOfContent = publicWidget.Widget.extend({
     },
 });
 
-publicWidget.registry.anchorSlide.include({
-
-    //--------------------------------------------------------------------------
-    // Private
-    //--------------------------------------------------------------------------
-
+patch(registry.category("website.active_elements").get("website.anchor_slide").prototype, {
```
Comment (ged-odoo) [inline addons/website/static/src/snippets/s_table_of_content/000.js]: you can import AnchorSlide directly
```js
import { AnchorSlide } from "@website/interactions/anchor_slide";

patch(AnchorSlide.prototype, { ... });
```

File: addons/website/static/tests/interactions/scroll_button.test.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import { expect, test } from "@odoo/hoot";
+import { animationFrame, click } from "@odoo/hoot-dom";
+import { startInteractions, setupInteractionWhiteList } from "../core/helpers";
+import { delay } from "@web/core/utils/concurrency";
+
+setupInteractionWhiteList("website.scroll_button");
+
+function isElementInViewport(el) {
```
Comment (ged-odoo) [inline addons/website/static/tests/interactions/scroll_button.test.js]: you can probably move this to helpers.js file in `../core/helpers` and reuse it in both tests

File: addons/website/static/tests/interactions/anchor_slide.test.js (hunk 2 above)
Comment (bso-odoo) [reply addons/website/static/tests/interactions/anchor_slide.test.js]: Nope, but it does not work with `AnchorSlide` either...
Edit: now relying on `advanceTime`

**Pull request merged**